tests/cupy_tests/indexing_tests/test_generate.py

```diff
@@ -24,3 +24,95 @@ def test_ix_empty_ndarray(self, xp):
     @testing.numpy_cupy_array_list_equal()
     def test_ix_bool_ndarray(self, xp):
         return xp.ix_(xp.array([True, False] * 2))
+
+
+@testing.gpu
+class TestR_(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_r_1(self, xp, dtype):
+        a = testing.shaped_arange((3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 4), xp, dtype)
+        return xp.r_[a, b]
+
+    def test_r_8(self, xp, dtype):
```

Comment:
`xp` is given only when decorated with `@testing.numpy_cupy_array_equal` etc. is given.